lib/dao.js

```diff
@@ -2187,7 +2187,7 @@ DataAccessObject.destroyAll = function destroyAll(where, options, cb) {
         var handleUndefined =  Model._getSetting('normalizeUndefinedInQuery');
         // alter configuration of how removeUndefined handles undefined values
         where = removeUndefined(where, handleUndefined);
-        where = Model._coerce(where, options);
+        where = Model._coerce(where);
```

Comment:
The `options` object are empty, therefore the test case failed in 2 occurrences.. 
once in juggler
```
 should count using date string
```
and once in Mongo 

```
mongodb default id type should be able to delete by string id
```
Where the `option` variable is not even affection the current fix as it is not needed..
The reason why , it seems that once the `Option` object is empty, it causes  an assertion failure, then the test case fail.